src/Fractions/Formatter/DefaultFractionFormatter.cs

```diff
@@ -7,102 +7,130 @@
-                ? (fraction.Numerator / fraction.Denominator).ToString(CultureInfo.InvariantCulture)
-                : BigInteger.Zero.ToString(CultureInfo.InvariantCulture);
+        if (fraction.IsPositiveInfinity) {
+            return numberFormatInfo.PositiveInfinitySymbol;
         }
 
-    private static string FormatRemainder(Fraction fraction) {
-            if (BigInteger.Abs(fraction.Numerator) < BigInteger.Abs(fraction.Denominator)) {
-                return FormatGeneral(fraction);
-            }
+        if (fraction.IsNegativeInfinity) {
+            return numberFormatInfo.NegativeInfinitySymbol;
+        }
+
+        return (fraction.Numerator / fraction.Denominator).ToString(formatProvider);
+    }
 
-            var integer = fraction.Denominator != BigInteger.Zero
-                ? fraction.Numerator / fraction.Denominator
-                : BigInteger.Zero;
+    private static string FormatRemainder(Fraction fraction, IFormatProvider formatProvider) {
+        if (fraction.Denominator.IsZero) {
+            return string.Empty;
+        }
 
-            var remainder = fraction - integer;
-            return FormatGeneral(remainder);
+        var isLessThanOne = BigInteger.Abs(fraction.Numerator) < BigInteger.Abs(fraction.Denominator);
+        if (isLessThanOne) {
+            return FormatGeneral(fraction, formatProvider);
         }
 
-    private static string FormatGeneral(Fraction fraction) {
-            if (fraction.Denominator == BigInteger.Zero) {
-                return BigInteger.Zero.ToString(CultureInfo.InvariantCulture);
+        var integer = fraction.Numerator / fraction.Denominator;
+        var remainder = fraction - integer;
+        return FormatGeneral(remainder, formatProvider);
+    }
+
+    private static string FormatGeneral(Fraction fraction, IFormatProvider formatProvider) {
+        var numberFormatInfo = (NumberFormatInfo)formatProvider.GetFormat(typeof(NumberFormatInfo))
+                               ?? CultureInfo.InvariantCulture.NumberFormat;
+
+        if (fraction.State == FractionState.IsNormalized) {
+            if (fraction.IsNaN) {
+                return numberFormatInfo.NaNSymbol;
             }
 
-            if (fraction.Denominator == BigInteger.One) {
-                return fraction.Numerator.ToString(CultureInfo.InvariantCulture);
+            if (fraction.IsPositiveInfinity) {
+                return numberFormatInfo.PositiveInfinitySymbol;
             }
 
-            return string.Concat(
-                fraction.Numerator.ToString(CultureInfo.InvariantCulture),
-                "/",
-                fraction.Denominator.ToString(CultureInfo.InvariantCulture));
+            if (fraction.IsNegativeInfinity) {
+                return numberFormatInfo.NegativeInfinitySymbol;
+            }
+
+            if (fraction.Denominator == BigInteger.One) {
+                return fraction.Numerator.ToString(formatProvider);
+            }
         }

```

Comment:
Note that for `InvariantCultureInfo` the symbols for these are actually longer than the "a/b"  (e.g. "PositiveInfinity").

Reply:
I think as a user I would rather see NaN than 0/0 (or ∞ for infinity) when calling `ToString()`. I introduced another breaking change - the default FormatProvider is now `CultureInfo.CurrentCulture`.

![image](https://github.com/danm-de/Fractions/assets/3825484/d676e72b-8781-4067-aa25-9195b994f6bc)

Reply:
Sure, that's what I would expect from the `DecimalNotationFormatter`. Personally, I don't mind it either way, I don't think there is any real reason to ever be parsing these anyway 😄

Reply:
Definitely a special case - but `double` has this feature.
![image](https://github.com/danm-de/Fractions/assets/3825484/e0624290-9789-4de4-97d5-585f11778f1c)